qiskit/circuit/parameterexpression.py

```diff
@@ -386,6 +386,8 @@ def __eq__(self, other):
         if isinstance(other, ParameterExpression):
             return (self.parameters == other.parameters
                     and self._symbol_expr.equals(other._symbol_expr))
+        elif isinstance(other, (float, int)):
+            return self._symbol_expr == other
         elif isinstance(other, numbers.Number):
             return (len(self.parameters) == 0
                     and complex(self._symbol_expr) == other)
```

Comment:
I think you can just remove the `len(self.parameters) == 0` constraint and that should fix all cases (not just int and float) right?
```suggestion
        elif isinstance(other, numbers.Number):
            return complex(self._symbol_expr) == other
```